### PR TITLE
Use GH actions for CI

### PR DESCRIPTION
**Pull request by mbrandonw**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,18 +4,13 @@ imports = \
 xcodeproj:
 	PF_DEVELOP=1 swift run xcodegen
 
-linux-main:
-	sourcery \
```
Comment (mbrandonw) [inline Makefile]: getting rid of sourcery while I'm here

File: Dockerfile [hunk 2]
```diff
@@ -6,4 +6,4 @@ COPY . ./
 
 RUN swift package resolve
 RUN swift package clean
-CMD swift test --parallel
+CMD swift test --parallel -v
```
Comment (mbrandonw) [inline Dockerfile]: how do you feel about verbose test logging?

Comment (stephencelis) [reply Dockerfile]: Sounds good to me. Didn't realize it was available.

File: Makefile [hunk 3]
```diff
@@ -4,18 +4,13 @@ imports = \
...
-		--sources ./Tests/ \
-		--templates ./.sourcery-templates/ \
-		--output ./Tests/ \
-		--args testimports='$(imports)' \
-		&& mv ./Tests/LinuxMain.generated.swift ./Tests/LinuxMain.swift
-
-test-linux: linux-main
+test-linux:
```
Comment (stephencelis) [inline Makefile]: This should still depend on `linux-main`, right?
```suggestion
test-linux: linux-main
```

Comment (stephencelis) [reply Makefile]: Oh, or maybe not...it still runs on Linux I guess, no?

Comment (mbrandonw) [reply Makefile]: Right. Maybe we need the difference of running Linux tests from macOS versus Linux, but I think this is ok

**Pull request merged**